Antonio

antoniojoboy opened the pull request.

antoniojoboy committed:
update branch

antoniojoboy committed:
added elemts to sidebar

antoniojoboy committed:
updated sidebar

added minimap and text

antoniojoboy committed:
did this work

antoniojoboy committed:
update map javascript

antoniojoboy committed:
updata map

Drewbi committed:
Delete test file

Drewbi committed:
Merge branch 'antonio' of https://github.com/Drewbi/buildings-in-landscape-uwa into antonio

Drewbi commented on src/map/index.js:
```diff
@@ -17,9 +17,12 @@ const map = new mapboxgl.Map({
 currentPos.id = 'current-marker'
 const currentMarker = new mapboxgl.Marker(currentPos)
-  .setLngLat([locations[0].positions.lon, locations[0].positions.lat])
+  .setLngLat([locations[0].positions.lon+100, locations[0].positions.lat+100])
```
I'd probably stay away from changing actual coordinates of the marker just because when you zoom in far enough the marker will be displaced and wont show the actually position.

Drewbi commented on src/map/map.css:
```diff
@@ -23,9 +23,10 @@
-  background-position-x: left 8px;
-  width: 30px;
-  height: 30px;
+  background-position-x: left 6px;
```
I'd recommend looking into this approach using offsets when initialising the current marker: https://docs.mapbox.com/mapbox-gl-js/api/markers/

This approach looks like it worked for x but we really want the bear to be standing where the position is instead of being placed centered onto the position.

Drewbi committed:
Merge fix

Drewbi committed:
Merge fix

Drewbi committed:
Merge branch 'antonio' of https://github.com/Drewbi/buildings-in-landscape-uwa into antonio

Drewbi committed:
Merge branch 'antonio' of https://github.com/Drewbi/buildings-in-landscape-uwa into antonio

Drewbi committed:
Merge branch 'antonio' of https://github.com/Drewbi/buildings-in-landscape-uwa into antonio

Drewbi committed:
Merge branch 'antonio' of https://github.com/Drewbi/buildings-in-landscape-uwa into antonio

Drewbi committed:
Merge branch 'antonio' of https://github.com/Drewbi/buildings-in-landscape-uwa into antonio

Drewbi commented on src/map/index.js:
```diff
@@ -17,9 +17,15 @@ const map = new mapboxgl.Map({
 const currentMarker = new mapboxgl.Marker(currentPos)
-  .setLngLat([locations[0].positions.lon, locations[0].positions.lat])
+  .setLngLat([
+    locations[0].positions.lon + 100,
```
I'd probably stay away from changing actual coordinates of the marker just because when you zoom in far enough the marker will be displaced and wont show the actually position.

antoniojoboy committed:
udpate map javascript

antoniojoboy committed:
added compass edited mapguy

added the compass to indicate which direction is north, and also edited the bear map icon guy so that it is standing on the marker itself and not beside it.

Drewbi commented on src/map/index.js:
```diff
@@ -10,22 +10,35 @@ const map = new mapboxgl.Map({
-  pitchWithRotate: false,
-  dragRotate: false
+  pitchWithRotate: true,
+  dragRotate: true
```
While this is pretty cool and on an actual map I'd be ok with it, having it on a tiny map isn't really necessary and if used accidentally would really disorientate someone. I think we'll just leave these off.

Drewbi commented on src/map/index.js:
```diff
@@ -10,22 +10,35 @@ const map = new mapboxgl.Map({
+  dragRotate: true
 })
 
+var popupOffsets = {
```
Is this used?

Drewbi commented on src/map/index.js:
```diff
@@ -10,22 +10,35 @@ const map = new mapboxgl.Map({
     markerElement.className = 'marker'
     const { lat, lon } = entry.positions
     const position = [lon, lat]
+    console.log('each position')
```
Remove console logs

Drewbi commented on src/map/index.js:
```diff
@@ -10,22 +10,35 @@ const map = new mapboxgl.Map({
   .setLngLat([locations[0].positions.lon, locations[0].positions.lat])
   .addTo(map)
 
+var nav = new mapboxgl.NavigationControl()
```
```suggestion
const nav = new mapboxgl.NavigationControl()
```
also consider using a more explicit name like navControl or mapControl

Drewbi commented on src/map/map.css:
```diff
@@ -1,36 +1,54 @@
+    background-position-y: centre;
+    width: 60px;
+    height: 60px;
+    z-index: 0;
```
The current marker should be above the map markers
```suggestion
    z-index: 1;
```

Drewbi commented on src/map/map.css:
```diff
@@ -1,36 +1,54 @@
+    background-size: contain;
+    position: sticky;
+    top: -9px;
+    background-position-x: centre;
```
Are these necessary anymore?
```suggestion
    background-position-x: centre;
```

Drewbi reviewed (changes requested):
Just a few cleanup things but good job on getting the positioning to work :)

antoniojoboy committed:
Merge branch 'master' into antonio

Drewbi commented:
It seems this doesn't change anything as I integrated your map changes already so I'm going to close this.
